### PR TITLE
Update fmtlib version to 9.1.0

### PR DESCRIPTION
**Pull request by canhld94**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
- Update fmtlib version to 9.1.0

Primary motivation is to add Ceph as submodule (https://github.com/ClickHouse/ClickHouse/issues/62821), but I think it's nice to have in general (e.g. `fmt::ostream_formatter`).

### Documentation entry for user-facing changes

- [x] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All NOT Required Checks
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: Integration Tests
- [ ]  Exclude: Stateless tests
- [ ]  Exclude: Stateful tests
- [ ]  Exclude: Performance tests
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with Aarch64
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
---
- [ ]  Do not test
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by Algunenano** [comment]
Please do not skip tests when you are doing global changes

**Comment by robot-clickhouse** [comment]
*This is an automated comment for commit de5258128e48061514a82bdfdc5368852d6a5062 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/64959/de5258128e48061514a82bdfdc5368852d6a5062/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>A Sync</td><td>If it fails, ask a maintainer for help</td><td>⏳ pending</td></tr>
<tbody>
</table>
<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>Checks correctness of the PR's body</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

**Review by Algunenano — COMMENTED** [review]
Is there any reason why you decided to update to 9.1 and not the latest (10.2)?

**Comment by canhld94** [comment]
> Is there any reason why you decided to update to 9.1 and not the latest (10.2)?

@Algunenano I tried 10.2 first, but then found that there's too many compile errors. The main reason is fmt has a break chance since v10.0:

> Removed deprecated implicit conversions for enums and conversions to primitive types for compatibility with std::format and to prevent potential ODR violations

We need to figure out where are implicit conversions in ClickHouse and fix it. Let see if I can do it quick.

**Comment by canhld94** [comment]
@Algunenano done for almost types, except the new `Int8`. I think it could be a bug in the new `fmtv10`:
- v9: good https://godbolt.org/z/e5rzjxbEb
- v10: compile error https://godbolt.org/z/fEdsc75rb
I created an issue in fmt upstream (https://github.com/fmtlib/fmt/issues/4007). I think that the issue could be having `()` in type definition. In the mean time, do you have any idea for work around this? Since you added this type before and have some work on its formatter.

**Comment by Algunenano** [comment]
> I think that the issue could be having () in type definition. In the mean time, do you have any idea for work around this? Since you added this type before and have some work on its formatter.

The only workaround I could find is an ugly rewrite of `fmt::detail::arg_mapper` via partial specialization and adding a method for `Int8`: [This ugly hack](https://godbolt.org/z/oqxM7Kz8f)

Not sure if there are better options.

File: src/Processors/Formats/Impl/BSONEachRowRowInputFormat.cpp [hunk 1]
```diff
@@ -806,7 +806,7 @@ bool BSONEachRowRowInputFormat::readRow(MutableColumns & columns, RowReadExtensi
         else
         {
             if (seen_columns[index])
-                throw Exception(ErrorCodes::INCORRECT_DATA, "Duplicate field found while parsing BSONEachRow format: {}", name);
+                throw Exception(ErrorCodes::INCORRECT_DATA, "Duplicate field found while parsing BSONEachRow format: {}", name.toView());
```
Comment (Algunenano) [inline src/Processors/Formats/Impl/BSONEachRowRowInputFormat.cpp:809]: I guess the changes in this file are not needed after fixing `StringRef` right?

File: src/Storages/Kafka/KafkaConsumer.h [hunk 2]
```diff
@@ -1,5 +1,7 @@
 #pragma once
 
+#include <cppkafka/topic_partition.h>
```
Comment (Algunenano) [inline src/Storages/Kafka/KafkaConsumer.h:3]: Keeping the alphabetical order would make me happier

**Pull request merged**